stan/math/prim/mat/fun/gp_exp_quad_cov.hpp

```diff
@@ -84,43 +89,37 @@ template <typename T_x, typename T_sigma, typename T_l>
 inline
     typename Eigen::Matrix<typename stan::return_type<T_x, T_sigma, T_l>::type,
                            Eigen::Dynamic, Eigen::Dynamic>
-    gp_exp_quad_cov(const std::vector<T_x>& x, const T_sigma& sigma,
-                    const std::vector<T_l>& length_scale) {
+    gp_exp_quad_cov(const std::vector<Eigen::Matrix<T_x, Eigen::Dynamic, 1>> &x,
+                    const T_sigma &sigma,
+                    const std::vector<T_l> &length_scale) {
   using std::exp;
 
-  check_positive("gp_exp_quad_cov", "marginal variance", sigma);
-  for (size_t n = 0; n < x.size(); ++n) {
-    check_not_nan("gp_exp_quad_cov", "x", x[n]);
-  }
-  for (size_t n = 0; n < length_scale.size(); ++n) {
-    check_positive("gp_exp_quad_cov", "length-scale", length_scale[n]);
-    check_not_nan("gp_exp_quad_cov", "length-scale", length_scale[n]);
-  }
+  size_t x_size = x.size();
+  size_t l_size = length_scale.size();
+  check_positive_finite("gp_exp_quad_cov", "magnitude", sigma);
+  check_positive_finite("gp_exp_quad_cov", "length scale", length_scale);
+  check_size_match("gp_exp_quad_cov", "x dimension", x[0].size(),
```

Comment:
Bug: this will fail if `x` is size 0. Please fix.

Reply:
Pushed a test example. Please run this test: `test/unit/math/prim/mat/fun/gp_exp_quad_cov_test.cpp`